src/java/org/apache/cassandra/db/streaming/ComponentManifest.java

```diff
@@ -30,16 +35,31 @@
 import org.apache.cassandra.db.TypeSizes;
 import org.apache.cassandra.io.IVersionedSerializer;
 import org.apache.cassandra.io.sstable.Component;
+import org.apache.cassandra.io.sstable.Descriptor;
 import org.apache.cassandra.io.util.DataInputPlus;
 import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.net.AsyncStreamingOutputPlus;
+import org.apache.cassandra.streaming.StreamManager;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+import static java.lang.Math.min;
 
 public final class ComponentManifest implements Iterable<Component>
```

Comment:
We could almost call this thing `ComponentStreamer` now :)

Reply:
I have moved the streaming code out of `ComponentManifest`..